src/util/mapbox.js

```diff
@@ -113,5 +129,13 @@ function parseUrl(url: string): UrlObject {
 
 function formatUrl(obj: UrlObject): string {
     const params = obj.params.length ? `?${obj.params.join('&')}` : '';
-    return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    const url = `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    return transformURL(url);
 }
+
+exports.resolveURL = function(url: string): string {
+    url = transformURL(url);
+    const a = window.document.createElement('a');
+    a.href = url;
+    return transformURL(a.href);
```

Comment:
@kkaefer - I just realized that this snuck through incorrectly. The intent was for it to be a post resolve transform only. I imagine that developers might want to take advantage of the transform, inspect, or validate fully resolved URLs. With a partial URL, I would find it harder to transform URLs if I don't know the host they might resolve to. This is useful when deploying code to different environments, or bundling a plugin where you don't always know what environment it will be running in.